features/functests/performance/performance.go

```diff
@@ -1,73 +1,214 @@
 package performance
 
 import (
+	"fmt"
 	"os/exec"
 	"strings"
+	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+
+	ocv1 "github.com/openshift/api/config/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/apimachinery/pkg/labels"
 
 	"github.com/openshift-kni/baremetal-deploy/features/functests/utils/clients"
-	"github.com/openshift-kni/baremetal-deploy/features/functests/utils/namespace"
-
-	k8sv1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-const testNamespace = "perfomancetest"
+const (
+	perfTestNamespace                       = "perfomancetest"
+	perfWorkerNodesLabel                    = "node-role.kubernetes.io/worker-rt="
+	perfMachineConfigDaemonContainer        = "machine-config-daemon"
+	perfClusterNodeTuningOperatorNamespaces = "openshift-cluster-node-tuning-operator"
+	perfSysctlTimeout                       = 240
+	perfSysctlPollInterval                  = 2
+)
 
 var mcKernelArguments = []string{"isolcpus"}
 
 var _ = Describe("performance", func() {
-	beforeAll(func() {
-		namespace.Create(testNamespace, clients.K8s)
-		namespace.Clean(testNamespace, clients.K8s)
+	// 00-tuned-network-latency.yaml related verification
+	var _ = Context("Network latency parameters adjusted by the Node Tuning Operator", func() {
+		It("Should contain configuration injected through the 00-tuned-network-latency.yaml template", func() {
+			sysctlMap := map[string]string{
+				"net.core.busy_read":              "50",
+				"net.core.busy_poll":              "50",
+				"net.ipv4.tcp_fastopen":           "3",
+				"kernel.numa_balancing":           "0",
+				"kernel.sched_min_granularity_ns": "10000000",
+				"vm.dirty_ratio":                  "10",
+				"vm.dirty_background_ratio":       "3",
+				"vm.swappiness":                   "10",
+				"kernel.sched_migration_cost_ns":  "5000000",
+			}
+			const perfNodeNetworkLatencyTuned = "openshift-node-network-latency"
+			_, err := clients.CntoConfig.TunedV1().Tuneds(perfClusterNodeTuningOperatorNamespaces).Get(perfNodeNetworkLatencyTuned,metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred(), "cannot find the Cluster Node Tuning Operator object "+ perfNodeNetworkLatencyTuned)
+
+			nodes := getListOfNodes()
+			execSysctlOnWorkers(nodes, sysctlMap)
+		})
 	})
 
-	var _ = Context("RT Kernel Arguments", func() {
-		It("Nodes should contain kernel arguments set by machine configuration", func() {
-			nodes, err := clients.K8s.CoreV1().Nodes().List(metav1.ListOptions{
-				LabelSelector: "node-role.kubernetes.io/worker-rt=",
-			})
-			Expect(err).ToNot(HaveOccurred())
-			for _, node := range nodes.Items {
+	// 11-pre-boot-worker-tuning.yaml related verification
+	var _ = Context("Pre boot tuning adjusted by the Machine Config Operator ", func() {
+		const perfWorkerPreBootTuning = "11-worker-pre-boot-tuning"
```

Comment:
where did "11-worker-pre-boot-tuning" came from ?  did you mean "11-worker-rt-kernel" ?

Reply:
notice that both rt kernel and pre boot scripts are set on the same machine config

Reply:
@yanirq , ah , forgot to rebase my stuff.